### PR TITLE
Update dg

### PR DESCRIPTION
**Pull request by LowJiaHao99**
Add GitHub and Telegram implementations

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -154,6 +154,70 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Telegram and GitHub attribute implementations
+
+<img src="images/ArchitectureDiagram.png" width="280" />
+
+The diagram above shows that a `Student` may or may not have a `Telegram` and a `Github`. Students with empty `GitHub` and `Telegram` are stored using `GitHub` and `Telegram` instantiated with empty strings as shown below
+
+<img src="images/StudentWithEmptyTelegramAndGitHub.png" width="280" />
```
Comment (jaysmyname) [inline docs/DeveloperGuide.md:163]: I think you mean the diagram below. Also, the image seems too small in the preview of the DG, maybe make the width larger?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -154,6 +154,70 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Telegram and GitHub attribute implementations
+
+<img src="images/ArchitectureDiagram.png" width="280" />
```
Comment (jaysmyname) [inline docs/DeveloperGuide.md:159]: What is the purpose of this diagram here?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -154,6 +154,70 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+>     String trimmedGitHub = gitHub.trim();
+>     if (!GitHub.isValidGitHub(trimmedGitHub)) {
+>         throw new ParseException(GitHub.MESSAGE_CONSTRAINTS);
+>     }
+>     return new GitHub(trimmedGitHub);
+> }
+> ```
+
+GitHub and Telegram objects instantiated with null inputs have a value of ""
+Here is a snippet for the constructor of Telegram. GitHub also have a similar format
```
Comment (jaysmyname) [inline docs/DeveloperGuide.md:187]: Add fullstops at the end?

**Review by jaysmyname — CHANGES_REQUESTED** [review]
Overall pretty good, just a few visual changes and grammatical corrections, once fixed should be GTM.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (LowJiaHao99) [reply docs/DeveloperGuide.md:163]: The multiplicity is used to illustrate that Student may or may not have a Telegram and a  GitHub.

Think Ill add this line to make things clearer

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (LowJiaHao99) [reply docs/DeveloperGuide.md:159]: The multiplicity is used to illustrate that Student may or may not have a Telegram and a  GitHub.

**Pull request merged**